app/src/main/java/com/vci/vectorcamapp/imaging/presentation/ImagingScreen.kt

```diff
@@ -47,54 +50,72 @@ fun ImagingScreen(
     }
 
     val pagerState = rememberPagerState(
-        initialPage = 0,
+        initialPage = state.capturedSpecimensAndInferenceResults.size,
         pageCount = { state.capturedSpecimensAndInferenceResults.size + 1 }
     )
 
-    HorizontalPager(
-        state = pagerState, modifier = modifier.fillMaxSize()
-    ) { page ->
-        when {
-            page < state.capturedSpecimensAndInferenceResults.size -> {
-                CapturedSpecimenOverlay(
-                    specimen = state.capturedSpecimensAndInferenceResults[page].specimen,
-                    inferenceResult = state.capturedSpecimensAndInferenceResults[page].inferenceResult,
-                    modifier = modifier
-                )
-            }
+    LaunchedEffect(state.capturedSpecimensAndInferenceResults.size) {
+        pagerState.scrollToPage(state.capturedSpecimensAndInferenceResults.size)
+        onAction(ImagingAction.InitializationComplete)
```

Comment:
InitializationComplete is not an appropriate name for this action. The initialization is happening only the first time this screen is loaded. However, this block of code runs every time state.capturedSpecimensAndInferenceResults.size changes. And every time it changes, it is not an initialization.